### PR TITLE
Add script for generating a test devnet

### LINKED ISSUE #149: Add shell script `gen-devnet` to generate the Anvil state
## 📚 Context

We want to simplify the generation of the Anvil state for the devnet.

## ✔️ Solution

Create a new Go binary under `cmd/gen-devnet` responsible for building the Anvil state. This binary will:
- Receive the following command line parameters:
  - DApp machine template hash (used to deploy the dapp)
  - Path to the anvil state file output
  - Path to the deployment info file (we need a file that contains the addresses of the deployed contracts)
- Download the rollups contracts (see [`pkg/contracts/generate/main.go`](https://github.com/cartesi/rollups-node/blob/next/pkg/contracts/generate/main.go))
- Start Anvil in the background with `--dump-state` (https://github.com/foundry-rs/foundry/tree/master/crates/anvil)
- Deploy the rollups contracts using forge (https://book.getfoundry.sh/forge/deploying)
- Call the authority/history/dapp factory to deploy the authority, history, and DApp contracts using cast (https://book.getfoundry.sh/cast/)
- Stop anvil

We will run this command later inside a Dockerfile. So, it can be assumed that all necessary tools (e.g., foundry) are installed.

### PR DESCRIPTION
**Pull request by marcelstanley**
Closes #149

### CODE REVIEW

**Review by gligneul — COMMENTED** [review]
This is starting to look good :)

File: scripts/devnet/lib/contracts.sh [hunk 1]
```diff
@@ -0,0 +1,168 @@
+#!/usr/bin/env sh
```
Comment (marcelstanley) [inline scripts/devnet/lib/contracts.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: scripts/devnet/lib/util.sh [hunk 2]
```diff
@@ -0,0 +1,30 @@
+#!/usr/bin/env sh
```
Comment (marcelstanley) [inline scripts/devnet/lib/util.sh:1]:
```suggestion
#!/usr/bin/env bash
```

File: scripts/devnet/gen-devnet.sh [hunk 3]
```diff
@@ -0,0 +1,363 @@
...
+
+   exit 0
+}
+
+################################################################################
+# Finish processing and clean up environment
+finish() {
+    verbose "finishing up"
+
+    if [ -n "$work_dir" ]; then
```
Comment (gligneul) [inline scripts/devnet/gen-devnet.sh:71]: If `check_error` gets called, this will never be executed, leaving the work dir behind.

Comment (gligneul) [reply scripts/devnet/gen-devnet.sh:71]: Not a problem for now because this script will be run inside docker build.

**Review by gligneul — APPROVED** [review]
This amount of bash makes me uneasy. We should spend little time improving this code because only a hacker like @marcelstanley can maintain it. We should move on, and I will create an issue to rewrite this in Go.

File: scripts/devnet/gen-devnet.sh (hunk 3 above)
Comment (marcelstanley) [reply scripts/devnet/gen-devnet.sh:71]: Not really. This trap should guarantee that `finish` is executed no matter what:

https://github.com/cartesi/rollups-node/pull/168/files#diff-39ace69d7418eeb86b4569890b7a7f56e6a334ee8b235d854c5dc3d4cae7389cR323

**Review by fmoura — APPROVED** [review]

**Pull request merged**